### PR TITLE
feat(es/parser): Add disallow_ambiguous_jsx_like option to parse mts and cts files like tsc and babel

### PR DESCRIPTION
**Pull request by nissy-dev**
**Description:** Add `disallow_ambiguous_jsx_like` option to parse mts and cts files like tsc and babel.

**BREAKING CHANGE:**

**Related issue (if exists):** Part of #4679 I also referred to https://github.com/babel/babel/pull/13838

### CODE REVIEW

**Comment by kdy1** [comment]
Can you rebase?

**Comment by nissy-dev** [comment]
I rebased!

File: crates/swc_ecma_parser/src/lib.rs [hunk 1]
```diff
@@ -275,6 +275,13 @@ impl Syntax {
             Syntax::Es(..) => true,
         }
     }
+
+    pub fn disallow_ambiguous_jsx_like(self) -> bool {
```
Comment (kdy1) [inline crates/swc_ecma_parser/src/lib.rs:279]: Please make this not public

**Review by kdy1 — CHANGES_REQUESTED** [review]

File: crates/swc_ecma_parser/src/lib.rs (hunk 1 above)
Comment (kdy1) [reply crates/swc_ecma_parser/src/lib.rs:279]:
```suggestion
   fn disallow_ambiguous_jsx_like(self) -> bool {
```

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc_ecma_parser --breaking

**Pull request merged**